### DISCUSSION TITLE
wallet_invokeSnap - adding an extra attribute to request fails

### DISCUSSION THREAD

**Discussion by javaspeak** [MetaMask/snaps#1892] [Q&A]
Code like this works:

```
await window.ethereum.request({
    method: 'wallet_invokeSnap',
    params: { snapId: defaultSnapOrigin, request: { method: 'authorizeUTUSnap' } },
  });

```
However if I add an extra request attribute it does not:

```
await window.ethereum.request({
    method: 'wallet_invokeSnap',
    params: { snapId: defaultSnapOrigin, request: { method: 'authorizeUTUSnap', token: '1234' } },
  });
```
  
 It complains that it does not like token:

`An error happened: Invalid JSON-RPC request: At path: token -- Expected a value of type `never`, but received: `"1234"`.`

How do we pass additional request properties from site to the snap?

I think am missing something very basic here.  

Thank you

**Comment by Montoya** [comment]
You are passing the params incorrectly, it should be a nested object with key "params". It should look like: 

```
await window.ethereum.request({
  method: 'wallet_invokeSnap',
  params: {
      snapId: defaultSnapOrigin,
      request: {
        method: 'authorizeUTUSnap',
        params: {
          token: '1234'
        }
      }
  },
});
